include/boost/math/distributions/beta.hpp

```diff
@@ -390,13 +390,47 @@ namespace boost
       }
       using boost::math::beta;
 
-      // Corner case: check_x ensures x element of [0, 1], but PDF is 0 for x = 0 and x = 1. PDF EQN:
+      // Corner cases: check_x ensures x element of [0, 1], but PDF is 0 for x = 0 and x = 1. PDF EQN:
       // https://wikimedia.org/api/rest_v1/media/math/render/svg/125fdaa41844a8703d1a8610ac00fbf3edacc8e7
-      if(x == 0 || x == 1)
+      if(x == 0)
       {
-        return RealType(0);
+        if (a == 1)
+        {
+          return 1 / beta(a, b);
+        }
+        else if (a < 1)
+        {
+          policies::raise_overflow_error<RealType>(function, nullptr, Policy());
+        }
+        else
+        {
+          return RealType(0);
+        }
+      }
+      else if (x == 1)
+      {
+        if (b == 1)
+        {
+          return 1 / beta(a, b);
+        }
+        else if (b < 1)
+        {
+          policies::raise_overflow_error<RealType>(function, nullptr, Policy());
+        }
+        else
+        {
+          return RealType(0);
+        }
+      }
+      
+      // Bounds checking
+      RealType return_val = ibeta_derivative(a, b, x, Policy());
+      if (return_val > b)
+      {
+        return_val = b;
```

Comment:
This doesn't look right to me - surely the PDF can be larger than the shape parameter?  Indeed can it not go to infinity at the endpoints?

Reply:
You're right. I was trying to catch the instance where we overshoot the endpoint (like in the linked issue). I can't think of any particularly good ways to do so.

Reply:
`test_beta_dist` uses 100 eps as the tolerance in the spot checks. I get under 40 eps above the endpoint from the linked issue locally.

Reply:
Nod.  The only endpoint-overshoot that can occur is when one of alpha and beta is exactly one, and then the endpoint is `1 / beta(alpha, beta)` and we currently return values very close to but either side of that.  Whether that's actually an issue I'm not sure.